htdocs/product/stock/class/entrepot.class.php

```diff
@@ -604,4 +616,51 @@ function initAsSpecimen()
         $this->country_id=1;
         $this->country_code='FR';
     }
+
+	function get_full_arbo() {
+		
+		 global $user,$langs,$conf;
+		 
+		 $TArbo = array($this->libelle);
+		 
+		 $id = $this->id;
+
+		 while(true) {
```

Comment:
If data were corrupted with a record with a parent that has the record or parent, we may have here an infinite loop that hangs the application. Can you introduce a counter that stop the loop if depth is higher than counter.
See example on build_path_from_id_categ function with $protection parameter that make look stop after the depth 1000.